libbpf-tools/trace_helpers.c

```diff
@@ -159,6 +166,134 @@ const struct ksym *ksyms__get_symbol(const struct ksyms *ksyms,
 	return NULL;
 }
 
+struct parts {
+	struct part *parts;
+	int parts_sz;
+	int parts_cap;
+	char *strs;
+	int strs_sz;
+	int strs_cap;
+};
+
+static int parts__add_part(struct parts *parts, const char *name,
+			   unsigned int dev)
+{
+	size_t new_cap, name_len = strlen(name) + 1;
+	struct part *part;
+	void *tmp;
+
+	if (parts->strs_sz + name_len > parts->strs_cap) {
```

Comment:
nit: I don't think you had to do all this complexity for a list of partitions, we don't expect to have tens of thousands of them, right? It's ok to do it the same way as kallsyms, but just doing straightforward strdup() and malloc() would work just fine and be simpler to maintain.

Reply:
Agree, I'll change to do `strdup()` and `malloc()`, thanks!